internal/datastore/crdb/keys.go

```diff
@@ -49,3 +55,23 @@ func prefix(s string) string {
 	}
 	return prefix
 }
+
+// overlapKeysFromContext reads the request-provided initial overlap key set
+// from the grpc request metadata.
+func overlapKeysFromContext(ctx context.Context) keySet {
```

Comment:
what happens if they neglect to give a key? do we fail-safe or fail-open?

Reply:
fail open - the point is that you only put the key on requests you care about, so that you can get `insecure` performance without them

Reply:
okay, can we document that more clearly in the flag then?